genConfig.py

```diff
@@ -0,0 +1,146 @@
+import json
+from pprint import pprint
+import math
+from copy import deepcopy
+from commonFunctions import make_sure_path_exists
+
+def scanCombinations(scanParams):
+    myParam = scanParams.keys()[0]
+
+    r = dict(scanParams)
+    del r[myParam]
+
+    list=[]
+    myList=[]
+
+    if len(r) is not 0:
+        list=scanCombinations(r)
+
+    min = scanParams[myParam][0]
+    max = scanParams[myParam][1]
+    if max < min:
+        tmpVal = max
+        max = min
+        min = tmpVal
+    entries = 2
+    if len(scanParams[myParam]) == 3:
+        entries = scanParams[myParam][2]
+
+    from numpy import linspace
+    if len(list) is 0:
+        for value in linspace(min,max,entries):
+            tmp = {}
+            tmp[myParam]=value
+            myList.append(tmp)
+    else:
+        for point in list:
+            for value in linspace(min,max,entries):
+                tmp = dict(point)
+                tmp[myParam]=value
+                myList.append(tmp)
+    return myList
+
+def consolidateListType(myList,intTypeList):
```

Comment:
Change name to ConsolidateListIntType or something like that, Make it clear that you are checking that the integer type variables are integers